pkg/scheduler/eventhandlers.go

```diff
@@ -299,60 +299,6 @@ func responsibleForPod(pod *v1.Pod, profiles profile.Map) bool {
 	return profiles.HandlesSchedulerName(pod.Spec.SchedulerName)
 }
 
-// skipPodUpdate checks whether the specified pod update should be ignored.
-// This function will return true if
-//   - The pod has already been assumed, AND
-//   - The pod has only its ResourceVersion, Spec.NodeName, Annotations,
-//     ManagedFields, Finalizers and/or Conditions updated.
-func (sched *Scheduler) skipPodUpdate(pod *v1.Pod) bool {
-	// Non-assumed pods should never be skipped.
-	isAssumed, err := sched.SchedulerCache.IsAssumedPod(pod)
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("failed to check whether pod %s/%s is assumed: %v", pod.Namespace, pod.Name, err))
-		return false
-	}
-	if !isAssumed {
-		return false
-	}
-
-	// Gets the assumed pod from the cache.
-	assumedPod, err := sched.SchedulerCache.GetPod(pod)
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("failed to get assumed pod %s/%s from cache: %v", pod.Namespace, pod.Name, err))
-		return false
-	}
-
-	// Compares the assumed pod in the cache with the pod update. If they are
-	// equal (with certain fields excluded), this pod update will be skipped.
-	f := func(pod *v1.Pod) *v1.Pod {
```

Comment:
I say we should migrate some of these to `isPodUpdated` in `scheduling_queue.go`.

The function is used to determine if an unschedulabe pod should be moved out into backoff or active.

@ahg-g? WDYT?

Reply:
yes, we can add `ManagedFields` and `Finalizers`. I am not sure about `Annotations`, and `NodeName` isn't going to trigger a queue update.

Reply:
wait, this comment hasn't been addressed

Reply:
also `p.Status.Conditions` please

Reply:
p.Status = v1.PodStatus{} has been assigned in isPodUpdated

Reply:
ah missed, that

/lgtm